dev/TEAL_opcodes.md

```diff
@@ -447,3 +448,26 @@ The `bnz` instruction opcode 0x40 is followed by two immediate data bytes which
 - Pops: *... stack*, any
 - Pushes: any, any
 - duplicate last value on stack
+
+## cons
+
+- Opcode: 0x50 
+- Pops: *... stack*, {[]byte A}, {[]byte B}
+- Pushes: []byte
+- pop two byte strings A and B and join them
+
+`cons` panics if the result would be greater than 4096 bytes
+
+## substring
+
+- Opcode: 0x51 {uint8 start position}{uint8 end position}
+- Pops: *... stack*, []byte
+- Pushes: []byte
+- pop a byte string X. For immediate values in 0..255 N and M: extract a range of bytes from it starting at N up to and including M, push the substring result
+
+## substring3
+
+- Opcode: 0x52 
+- Pops: *... stack*, {[]byte A}, {uint64 B}, {uint64 C}
+- Pushes: []byte
+- pop a byte string A and two integers B and C. Extract a range of bytes from A starting at B up to and including C, push the substring result
```

Comment:
I think this shouldn't be inclusive on `C` in order to make iterating over substrings more natural. That way as I'm iterating over substrings the second argument becomes the first, like: `"foo".substring(0, 1)` and then `"foo".substring(1, 2)`

This is how slicing works in python and how the substring function works in javascript